Sources/MongoSwift/MongoCollection+BulkWrite.swift

```diff
@@ -393,21 +433,30 @@ public struct BulkWriteResult {
         if let upserted = try reply.getValue(for: "upserted") as? [Document] {
             for upsert in upserted {
                 guard let index = try upsert.getValue(for: "index") as? Int else {
-                    throw MongoError.typeError(message: "Could not cast upserted index to `Int`")
+                    throw RuntimeError.internalError(message: "Could not cast upserted index to `Int`")
                 }
                 upsertedIds[index] = upsert["_id"]
             }
         }
 
         self.upsertedIds = upsertedIds
+    }
 
-        if let writeErrors = try reply.getValue(for: "writeErrors") as? [Document] {
-            self.writeErrors = try writeErrors.map { try BSONDecoder().decode(BulkWriteError.self, from: $0) }
-        }
-
-        if let writeConcernErrors = try reply.getValue(for: "writeConcernErrors") as? [Document],
-            writeConcernErrors.indices.contains(0) {
-            self.writeConcernError = try BSONDecoder().decode(WriteConcernError.self, from: writeConcernErrors[0])
-        }
+    /// Internal initializer used for testing purposes and error handling.
+    internal init(
+            deletedCount: Int? = nil,
+            insertedCount: Int? = nil,
+            insertedIds: [Int: BSONValue]? = nil,
+            matchedCount: Int? = nil,
+            modifiedCount: Int? = nil,
+            upsertedCount: Int? = nil,
+            upsertedIds: [Int: BSONValue]? = nil) {
+        self.deletedCount = deletedCount ?? 0
```

Comment:
would it make sense to just use all of these default values as the defaults for the parameters, rather than setting new default values inside here?

Reply:
I would still have to unwrap them even if they had default values. I could force unwrap and add a swiftlint ignore if you think that's preferable.